MailCore/Cache/AccountManager/AccountManager.swift

```diff
@@ -384,51 +400,79 @@ public final class AccountManager: RefreshTokenDelegate, ObservableObject {
 
     public func switchMailbox(newMailbox: Mailbox) {
         Task {
-            setCurrentMailboxForCurrentAccount(mailbox: newMailbox, refresh: false)
-            saveAccounts()
+            self.setCurrentMailboxForCurrentAccount(mailbox: newMailbox, refresh: false)
+            self.saveAccounts()
+
+            guard let mailboxManager = getMailboxManager(for: newMailbox) else {
+                logError(.mailboxManagerMissing)
```

Comment:
eventualy we should have a proper error handling and not silently return in a guard (a pattern I saw in this file ~10 times)

Reply:
Definitely agree